components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediator.java

```
@@ -0,0 +1,390 @@
+                    messageContext.getEnvelope().getBody().addChild(omElement);
+                }
+            }
+        } catch (DataServiceFault dataServiceFault) {
+            if (synLog.isTraceOrDebugEnabled()) {
+                synLog.traceOrDebug(dataServiceFault.getMessage());
+            }
+            synLog.error(dataServiceFault.getMessage());
+            handleException("DataService exception occurred while accessing the dataservice to do the operation",
+                    dataServiceFault, messageContext);
+
+        }
+    }
+
+    private void addOperations(Operations operations, OMElement payload, MessageContext msgCtx) {
+
+        for (Object operationObj : operations.getOperations()) {
+            if (operationObj instanceof Operation) {
+                Operation nestedOperation = (Operation) operationObj;
+                String operationName = nestedOperation.getOperationName();
+                OMElement nestedPayload = fac.createOMElement(operationName, omNamespace);
+                for (Param param : nestedOperation.getParams()) {
+                    addParams(param, nestedPayload, msgCtx);
+                }
+                payload.addChild(nestedPayload);
+            } else if (operationObj instanceof Operations) {
+                Operations rootOperations = (Operations) operationObj;
+                String rootOpName;
+                if (rootOperations.getType() == OperationsType.SINGLE_REQ) {
+                    Operation singleOp = (Operation) rootOperations.getOperations().get(0);
+                    rootOpName = singleOp.getOperationName();
+                } else {
+                    rootOpName = rootOperations.getType().toString().toLowerCase();
+                }
+                OMElement nestedOpEle = fac.createOMElement(rootOpName, omNamespace);
+                addOperations(rootOperations, nestedOpEle, msgCtx);
+                payload.addChild(nestedOpEle);
+            }
```

else we need to throw unsupported type ?

---

From the factory class it adds only the Operation and Operations objects. So no need to handle it from here.